src/main/java/edu/harvard/iq/dataverse/api/Admin.java

```diff
@@ -1234,4 +1244,135 @@ public Response clearMetricsCacheByName(@PathParam("name") String name) {
 		return ok("metric cache " + name + " cleared.");
 	}
 
+    @GET
+    @Path("/dataverse/{alias}/addAdminsToChildren")
+    public Response addAdminsToChildren(@PathParam("alias") String alias) throws WrappedResponse {
+        Dataverse owner = dataverseSvc.findByAlias(alias);
+        if (owner == null) {
+            return error(Response.Status.NOT_FOUND,
+                    "Could not find dataverse based on alias supplied: " + alias + ".");
+        }
+        try {
+            AuthenticatedUser user = findAuthenticatedUserOrDie();
+            if (!user.isSuperuser()) {
+                return error(Response.Status.FORBIDDEN, "Superusers only.");
+            }
+        } catch (WrappedResponse wr) {
+            return wr.getResponse();
+        }
+        if (settingsSvc.isTrueForKey(SettingsServiceBean.Key.InheritParentAdmins, false)) {
+
+            String qstr = "WITH RECURSIVE path_elements AS ((" + " SELECT id, dtype FROM dvobject WHERE id in ("
```

Comment:
What does this query do?

Reply:
Added comments in the code to explain the query...